src/SprykerEco/Zed/AmazonQuicksight/Business/Validator/QuicksightAnalyticsRequestValidator.php

```diff
@@ -0,0 +1,157 @@
+     *
+     * @return \Generated\Shared\Transfer\EnableQuicksightAnalyticsResponseTransfer
+     */
+    public function validateEnableQuicksightAnalyticsRequest(
+        EnableQuicksightAnalyticsRequestTransfer $enableQuicksightAnalyticsRequestTransfer,
+        EnableQuicksightAnalyticsResponseTransfer $enableQuicksightAnalyticsResponseTransfer
+    ): EnableQuicksightAnalyticsResponseTransfer {
+        $quicksightAssetBundleImportJobTransfer = $enableQuicksightAnalyticsRequestTransfer->getQuicksightAssetBundleImportJob();
+
+        if (!$this->isEnableAnalyticsEnabled($quicksightAssetBundleImportJobTransfer)) {
+            $enableQuicksightAnalyticsResponseTransfer->addError(
+                (new ErrorTransfer())->setMessage(static::ERROR_MESSAGE_ENABLE_ANALYTICS_FAILED),
+            );
+        }
+
+        return $enableQuicksightAnalyticsResponseTransfer;
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\ResetQuicksightAnalyticsRequestTransfer $resetQuicksightAnalyticsRequestTransfer
+     * @param \Generated\Shared\Transfer\ResetQuicksightAnalyticsResponseTransfer $resetQuicksightAnalyticsResponseTransfer
+     *
+     * @return \Generated\Shared\Transfer\ResetQuicksightAnalyticsResponseTransfer
+     */
+    public function validateResetQuicksightAnalyticsRequest(
+        ResetQuicksightAnalyticsRequestTransfer $resetQuicksightAnalyticsRequestTransfer,
+        ResetQuicksightAnalyticsResponseTransfer $resetQuicksightAnalyticsResponseTransfer
+    ): ResetQuicksightAnalyticsResponseTransfer {
+        $quicksightAssetBundleImportJobTransfer = $resetQuicksightAnalyticsRequestTransfer->getQuicksightAssetBundleImportJob();
+        $quicksightUserTransfer = $resetQuicksightAnalyticsRequestTransfer->getUserOrFail()->getQuicksightUser();
+
+        if (!$this->isResetAnalyticsEnabled($quicksightAssetBundleImportJobTransfer, $quicksightUserTransfer)) {
+            $resetQuicksightAnalyticsResponseTransfer->addError(
+                (new ErrorTransfer())->setMessage(static::ERROR_MESSAGE_RESET_ANALYTICS_FAILED),
+            );
+        }
+
+        return $resetQuicksightAnalyticsResponseTransfer;
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\QuicksightAssetBundleImportJobTransfer|null $quicksightAssetBundleImportJobTransfer
+     * @param \Generated\Shared\Transfer\QuicksightUserTransfer|null $quicksightUserTransfer
+     *
+     * @return bool
+     */
+    public function isResetAnalyticsEnabled(
+        ?QuicksightAssetBundleImportJobTransfer $quicksightAssetBundleImportJobTransfer,
+        ?QuicksightUserTransfer $quicksightUserTransfer
+    ): bool {
+        return $this->isAssetBundleSuccessfullyInitialized($quicksightAssetBundleImportJobTransfer)
+            && !$this->isAssetBundleInitializationInProgress($quicksightAssetBundleImportJobTransfer)
+            && $quicksightUserTransfer && $quicksightUserTransfer->getRole() === static::QUICKSIGHT_USER_ROLE_AUTHOR;
```

Comment:
Does it make sense to move these two checks to a separate method for consistency?